brave/src/main/java/brave/handler/SpanCollector.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.handler;
+
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.TraceContext;
+import java.lang.ref.WeakReference;
+
+/**
+ * This tracks one recording of a {@link TraceContext}. Common implementations include span
+ * reporting (ex to Zipkin) and data manipulation, such as redaction for security purposes.
+ *
+ * <h3>Relationship to Span lifecycle</h3>
+ * The pair of {@link #begin} and {@link #end} seems the same as the span lifecycle. In
+ * most cases it will be the same, but you cannot assume this.
+ *
+ * <p>A {@link TraceContext} could be recorded twice, for example, if a long operation
+ * began, called {@link Span#flush()} (recording 1) and later called {@link Span#finish()}
+ * (recording 2). A {@link TraceContext} could be abrupted by garbage collection resulting in a
+ * {@link Cause#ABANDON}. A user could even {@linkplain Cause#ABANDON abandon} a span without
+ * recording anything!
+ *
+ * <p>Collectors that process finished spans will need to look at the {link Cause} and {@link
+ * MutableSpan} collected. For example, {@link Cause#FINISH} is usually a good enough heuristic to
+ * find complete spans.
+ *
+ * <h3>Advanced Notes</h3>
+ * <p>It is important to do work quickly as callbacks are run on the same thread as application
+ * code. However, do not mutate {@link MutableSpan} between callbacks, as it is not thread safe.
+ *
+ * <p>The {@link TraceContext} and {@link MutableSpan} parameter from {@link #begin} will be
+ * the same reference for {@link #end}.
+ *
+ * <p>If caching the context or span parameters between callbacks, consider a {@link WeakReference}
+ * to avoid holding up garbage collection.
+ *
+ * <p>The {@link #begin} callback primarily supports tracking of children, or partitioning of
+ * data for backend that needs to see an entire {@linkplain TraceContext#localRootId() local root}.
+ */
+public interface SpanCollector {
+  /** What ended the data collection? */
+  enum Cause {
```

Comment:
for an enum `Cause` i would linguistically expect past-tense constants, `ABANDONED` - `FINISHED` etc ?

Reply:
sgtm thanks